docs/DeveloperGuide.adoc

```diff
@@ -295,62 +295,87 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 [width="59%",cols="22%,<23%,<25%,<30%",options="header",]
 |=======================================================================
 |Priority |As a ... |I want to ... |So that I can...
+|`* * *` |data-oriented person |see an overview of my transactions in an intuitive graph |gain insights at a glance
+
 |`* * *` |new user |see usage instructions |refer to instructions when I forget how to use the App
 
-|`* * *` |user |add a new person |
+|`* * *` |student |input my spending into different categories |manage my expenses better
+
+|`* * *` |visual person | see how much budget I have left in each category |cut down on spending as necessary
+
+|`* * *` |student with many friends |split the bill with my friends |know how much I should pay for a meal
 
-|`* * *` |user |delete a person |remove entries that I no longer need
+|`* * *` |student who forgets to pay his debt on time |be reminded to pay my debt before the deadline |stop incurring interests
 
-|`* * *` |user |find a person by name |locate details of persons without having to go through the entire list
+|`* *` |patriotic Singaporean who travels to JB often |easily calculate how much Ringgit to bring |enjoy my holiday with insufficient money
 
-|`* *` |user |hide <<private-contact-detail,private contact details>> by default |minimize chance of someone else seeing them by accident
+|`* *` |thrifty person |see how my savings or spending will project into the future |plan my budget
+
+|`*` |user who forgets to save money |be incentivized me to save money |be motivated to not overspend/save money
 
-|`*` |user with many persons in the address book |sort persons by name |locate a person easily
 |=======================================================================
 
 _{More to be added}_
 
 [appendix]
 == Use Cases
 
-(For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `PalPay` and the **Actor** is the `user`, unless specified otherwise)
 
 [discrete]
-=== Use case: Delete person
+=== Use case: Add an income
 
 *MSS*
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1. User requests to add an income amount.
+2. PalPay adds the income amount.
 +
 Use case ends.
 
 *Extensions*
 
 [none]
-* 2a. The list is empty.
+* 1a. Amount entered by the user is invalid.
++
+[none]
+** 1a1. PalPay shows an error message.
++
+Use case resumes at step 1.
+
+[discrete]
+=== Use case: Delete expense
+
+*MSS*
+
+1. User requests to view spending.
+2. PalPay shows the list of expenses since beginning of time.
+3. User requests to delete an expense.
+4. PalPay deletes the specified expense.
 +
 Use case ends.
 
-* 3a. The given index is invalid.
+*Extensions*
```

Comment:
It might be better to include more exceptional cases for situations where the user enters incorrect details.